### PR TITLE
Add input validation to identity settings

### PR DESCRIPTION
**Pull request by alexhauser**
Issue #457.

**Description:**

Add input validation to all `EditText` input fields in the identity settings activtiy. This fixes a bunch of issues which could be caused by either leaving input fields empty or entering very large values.

**Changes:**

- Before, the few validations present were done upon hitting the "Save" button. Now, input validation is done "on-the-fly", directly when entering the values.
- The "Save" button is now dynamically enabled/disabled according to the input validation state
- Input errors are displayed directly on the affected input field instead of in a popup message
- Remove the arbitrary _QuickPass Timeout_ limit of one week

The whole process should be much cleaner now.

**Validations that are performed now:**

- Check for empty field
- Check for non-numeric values
- Upper bounds checking

### CODE REVIEW

File: app/src/main/java/org/ea/sqrl/activites/identity/IdentitySettingsActivity.java [hunk 1]
```diff
@@ -110,25 +97,83 @@ public void onResume() {
...
-        cbSettingsNoBypass.setChecked(storage.isNoByPass());
+        txtQuickPassLength.setText(Integer.toString(storage.getHintLength()));
+        txtPwdVerifySecs.setText(Integer.toString(storage.getPasswordVerify()));
+        txtQuickPassTimeout.setText(Integer.toString(storage.getIdleTimeout()));
+        cbSQRLOnly.setChecked(storage.isSQRLOnly());
+        cbNoBypass.setChecked(storage.isNoByPass());
     }
 
-    public int getIntValue(EditText txt, int errorMessage) {
+    public int getIntValue(EditText txt) {
```
Comment (kalaspuffar) [inline app/src/main/java/org/ea/sqrl/activites/identity/IdentitySettingsActivity.java]: Is -1 a valid number? It is below 65535 at least.

Comment (alexhauser) [reply app/src/main/java/org/ea/sqrl/activites/identity/IdentitySettingsActivity.java]: This method always returned -1 on error, I haven't changed anything there. I've just removed the call for showing the error message. Or am I missing your point?

Comment (alexhauser) [reply app/src/main/java/org/ea/sqrl/activites/identity/IdentitySettingsActivity.java]: If you're talking about lower bounds checking on the input fields, that would need to happen in `validateInputFields()`. And yes, I've thought about it too. I guess we should only allow positive integers. Right? If you agree, I'll modify my code to accomodate that.

Comment (kalaspuffar) [reply app/src/main/java/org/ea/sqrl/activites/identity/IdentitySettingsActivity.java]: Yes, that was what I was thinking about. Might have expressed it a bit bad.

**Comment by alexhauser** [comment]
Added lower bounds check - please review again, @kalaspuffar. 

It technically isn't really necessary, since the input field doesn't let you input negative numbers, but hey - belts and suspenders. 😀

**Pull request merged**